File: frontend/src/models/Activity.js
export const ActivityStatus = {
  Planned: 'Planned',
  Reserved: 'Reserved',
  Completed: 'Completed',
  Cancelled: 'Cancelled',
};

export class Activity {
  constructor({ id, tripId, name, date, time, location, description, estimatedCost, status }) {
    this.id = id;
    this.tripId = tripId;
    this.name = name;
    this.date = date;
    this.time = time;
    this.location = location;
    this.description = description;
    this.estimatedCost = estimatedCost;
    this.status = status ?? ActivityStatus.Planned;
  }
}

File: TripService/Models/Activity.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TripService.Models
{
    public enum ActivityStatus { Planned, Reserved, Completed, Cancelled }

    public class Activity
    {
        public int Id { get; set; }
        public int TripId { get; set; }

        [Column("Title")]
        public required string Name { get; set; }

        [Column("ActivityDate")]
        public DateTime Date { get; set; }

        public TimeSpan Time { get; set; }
        public required string Location { get; set; }
        public string Description { get; set; } = string.Empty;
        public decimal EstimatedCost { get; set; }
        public ActivityStatus Status { get; set; } = ActivityStatus.Planned;

        public Trip Trip { get; set; } = null!;
    }
}
